src/states/node.rs

```diff
@@ -479,24 +490,15 @@ impl Node {
                 return;
             }
         } else {
```

Comment:
We can combine the `if` below into a single line `else if` instead of an explicit split done here consuming 2 more lines of code and not buying anything (i.e. has no differing side-effect)